scripts/sct_deepseg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""
+This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+ivadomed package.
+"""
+
+# TODO: Add link to example image so users can decide wether their images look "close enought" to some of the proposed
+#  models (e.g., mice, etc.).
+# TODO: add test to make sure that all postprocessing flags match the core.DEFAULT dictionary items
+# TODO: Fetch default value (and display) depending on the model that is used.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+import colored
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.core
+import spinalcordtoolbox.deepseg.models
+import spinalcordtoolbox.utils
+
+from sct_utils import init_sct, printv, display_viewer_syntax
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=sct.utils.SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        help="Image to segment.",
+        metavar=sct.utils.Metavar.file)
+
+    seg = parser.add_argument_group('\nMODELS')
+    seg.add_argument(
+        "-m",
+        help="Model to use. For a description of each available model, please write: 'sct_deepseg -list-models'.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-list-models",
+        action='store_true',
+        help="Display a list of available models.")
+    seg.add_argument(
+        "-install-model",
+        help="Install specified model.",
+        choices=list(sct.deepseg.models.MODELS.keys()))
+    seg.add_argument(
+        "-install-default-models",
+        action='store_true',
+        help="Install all default models. Note: default models should already be installed during SCT installation.")
+    # TODO: replace URL below
+    seg.add_argument(
+        "-mpath",
+        help="Path to model, in case you would like to use a custom model. The model folder should follow the "
+             "conventions listed in: URL.",
```

Comment:
I think you should roll `-m` and `-mpath` together. Just provide `-m` but decide based on what is given whether you're using an explicit path to a model or one of the ones from `__models_dir__`. It's confusing from the user's perspective to have two things that mean the same thing but are subtly different.

You could do this the way $PATH works: given `-m $MODEL` you look for `$DIR/$MODEL for $DIR in $MODELS_PATH` (--and I guess just `$MODEL` by itself) and pick the first one that wins. That way, someone can give `-m /mnt/nfs/neuromodels/custom-brain-model` or `-m ./custom-model` and it will just work.

Reply:
Good idea. Implemented in 3588ad354f535bfded2d972b7ab51509bd9b6cc7